File: integration/presto/src/test/java/org/apache/carbondata/presto/it/PrestoIntegrationTest.java
package org.apache.carbondata.presto.it;

import org.junit.BeforeClass;
import org.junit.Test;


public class PrestoIntegrationTest {
    private static String CARBONDATA_STOREPATH = "$rootPath/integration/presto/test/store";
    private static org.apache.carbondata.presto.PrestoServer prestoServer;
    @BeforeClass
    public static void setUp() throws Exception {
        StoreCreator.createCarbonStore();
        prestoServer = new org.apache.carbondata.presto.PrestoServer();
        prestoServer.prestoJdbcClient(CARBONDATA_STOREPATH );
    }
    @Test
    public void demo(){

    }
}

File: integration/presto/src/test/scala/org/apache/carbondata/presto/it/PrestoIntegrationTest.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.carbondata.presto.it

import java.io.File

import org.scalatest.{BeforeAndAfterAll, FunSuiteLike}
import util.StoreCreator

import org.apache.carbondata.common.logging.LogServiceFactory
import org.apache.carbondata.presto.server.PrestoServer


class PrestoIntegrationTest extends FunSuiteLike with BeforeAndAfterAll {

  private val logger = LogServiceFactory
    .getLogService(classOf[PrestoIntegrationTest].getCanonicalName)

  private val rootPath = new File(this.getClass.getResource("/").getPath
                          + "../../../..").getCanonicalPath
  private val storePath = s"$rootPath/integration/presto/target/store"

  override def beforeAll: Unit = {
    StoreCreator.createCarbonStore(storePath)
    logger.info(s"\nCarbon store is created at location: $storePath")
    PrestoServer.startServer(storePath)
  }

  test("test the fetch queries in presto") {
   println(PrestoServer.executeQuery("SELECT * FROM TESTDB.TESTTABLE"))
  }

  override def afterAll: Unit = {

  }
}